packages/cohort/src/bin/replicator.rs

```diff
@@ -1,15 +1,76 @@
-use cohort::replicator::{
-    core::{Replicator, ReplicatorCandidate, StatemapItem},
-    replicator_service::run_talos_replicator,
+use std::{io::Error, sync::Arc, vec};
+
+use cohort::{
+    bank_api::Transfer,
+    config_loader::ConfigLoader,
+    replicator::{
+        core::{Replicator, ReplicatorCandidate, StatemapItem},
+        replicator_service::run_talos_replicator,
+    },
+    state::{
+        model::AccountRef,
+        postgres::database::{Database, StatementsAndParams},
+    },
 };
 use log::info;
+use serde::{Deserialize, Serialize};
 use talos_certifier::ports::MessageReciever;
 use talos_certifier_adapters::{KafkaConfig, KafkaConsumer};
 use talos_suffix::{core::SuffixConfig, Suffix};
+use tokio_postgres::types::ToSql;
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ReplicatorBankTransfer {
+    from: String,
+    to: String,
+    amount: f64,
+}
+
+async fn statemap_install_handler(sm: Vec<StatemapItem>, db: Arc<Database>) -> Result<bool, Error> {
+    info!("Original statemaps received ... {:#?} ", sm);
+
+    let mut client = db.pool.get().await.unwrap();
+    let transaction = client.transaction().await.unwrap();
+
+    // Convert the StatemapItems to the one understood by BankAPI.
+    let bank_batch: Vec<Box<dyn StatementsAndParams>> = sm.into_iter().fold(vec![], |mut acc, sm_item| {
+        if sm_item.action == "bank_transfer" {
+            let new_version = sm_item.version;
+            let payload = serde_json::from_value::<ReplicatorBankTransfer>(sm_item.payload).unwrap();
+
+            acc.push(Box::new(Transfer::new(
+                AccountRef::new(payload.from, Some(new_version)),
+                AccountRef::new(payload.to, Some(new_version)),
+                payload.amount.to_string(),
+                new_version,
+                true,
```

Comment:
this should be `true` only for the last item in the list because SQL statements are written something like that

UPDATE x=..., version=v WHERE version < v; 
UPDATE x=..., version=v WHERE version < v;  // this will not execute skipped if line above updated 'v'.

Reply:
But you dont worry about it, in the new batch api I am handling this internally.